pkg/pullsync/pullsync.go

```diff
@@ -87,21 +88,23 @@ func New(
 	streamer p2p.Streamer,
 	store storer.Reserve,
 	unwrap func(swarm.Chunk),
+	gsocHandler func(soc.SOC),
```

Comment:
This handler function should take a pointer to SOC. Let's change this everywhere

Reply:
why is it need a pointer? I thought a struct value is already a reference.

in any way, I took the hook function of PSS as a sample above this line.